File: Java/Move_Zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int k = 0;
        for (int i : nums) {
            if (i != 0) {
                nums[k] = i;
                k++;
            }
        }
        for (int i = k; i < nums.length; i++) {
            nums[i] = 0;
        }
    }
}

File: Kotlin/Move_zeroes.kt
class Solution {
    fun moveZeroes(nums: IntArray) {
        var k = 0
        for (i in nums.indices) {
            if (nums[i] != 0) {
                nums[k++] = nums[i]
            }
        }
        for (i in k until nums.size) {
            nums[i] = 0
        }
    }
}
